crates/node/src/txvalidation/event.rs

```diff
@@ -202,3 +273,275 @@ impl TxEvent<NewTx> {
             .await
     }
 }
+
+// Use to cache New tx and store waiting Tx that have missing parent.
+pub struct TXCache {
+    // List of Tx waiting for parent.let waiting_txs =
+    waiting_tx: HashMap<Hash, (Vec<TxEvent<WaitTx>>, Instant)>,
+    // Cache of the last saved Tx in the DB. To avoid to query the db for Tx.
+    cachedtx_for_verification: LruCache<Hash, WaitTx>,
+    // Number of Waiting Tx that trigger the Tx eviction process.
+    max_waiting_tx: usize,
+    // Max time a Tx can wait in millisecond.
+    max_waiting_time: Duration,
+}
+
+impl TXCache {
+    pub fn new() -> Self {
+        Self::build(
+            MAX_CACHED_TX_FOR_VERIFICATION,
+            MAX_WAITING_TX_FOR_VERIFICATION,
+            MAX_WAITING_TIME_IN_MS,
+        )
+    }
+    pub fn build(cache_size: usize, max_waiting_tx: usize, max_waiting_time: u64) -> Self {
+        let cachedtx_for_verification =
```

Comment:
```suggestion
        let cached_tx_for_verification =
```
Would it be ok to separate the words like this so it's easier to read when skimming the code through?

Reply:
I've updated all occurrences in the code.